surface/model_openapiv3.go

```diff
@@ -185,40 +233,75 @@ func (b *OpenAPI3Builder) buildTypeFromParameters(
 			}
 			f.Serialize = true
 			t.addField(&f)
+		} else if parameterRef := parametersItem.GetReference(); parameterRef != nil {
+			f.Type = typeForRef(parameterRef.GetXRef())
+			f.Name = strings.ToLower(f.Type)
+			f.Serialize = false
+			f.Kind = FieldKind_REFERENCE
+			t.addField(&f)
 		}
 	}
+
+	_, err = b.buildTypeFromRequestBody(name, requestBody, t)
+
+	if len(t.Fields) > 0 {
+		b.model.addType(t)
+		return t.Name, err
+	}
+	return "", err
+}
+
+// buildTypeFromRequestBody builds a service type description from the request bodies of an OpenAPI
+// description. If tIn is not given, a new type is created. Otherwise tIn is used.
+func (b *OpenAPI3Builder) buildTypeFromRequestBody(name string, requestBody *openapiv3.RequestBodyOrReference, tIn *Type) (tOut *Type, err error) {
+	if tIn == nil {
+		tOut = &Type{}
+		tOut.Name = name
+	} else {
+		tOut = tIn
+	}
+
 	if requestBody != nil {
 		content := requestBody.GetRequestBody().GetContent()
+		var f Field
```

Comment:
```go
f := Field{
  Position: Position_BODY,
  Serialize: true,
}
```

Reply:
I think I see some other spots in the original code (not what is changed in this PR) that uses the style shown in this PR. So I understand following that as convention. But I believe initializing via a composite literal is the "Go way".

Reply:
Done. I also changed it for the other spots in the original code to keep it more consistent